src/lib/libemval.js

```diff
@@ -109,6 +110,85 @@ var LibraryEmVal = {
     for (var i = 0; i < view.length; i++) a[i] = view[i];
     return Emval.toHandle(a);
   },
+  _emval_array_to_memory_view__deps: ['$Emval'],
+  _emval_array_to_memory_view: (dst, src) => {
+    dst = Emval.toValue(dst);
+    src = Emval.toValue(src);
+    dst.set(src);
+  },
+#else
+  _emval_new_array_from_memory_view__deps: ['$Emval'],
+  _emval_new_array_from_memory_view: (view) => {
+    view = Emval.toValue(view);
+    var reader = (() => {
+      if (view.BYTES_PER_ELEMENT===1)
+        return i => view[i];
+      const dv = new DataView(view.buffer, view.byteOffset);
+      switch(view.BYTES_PER_ELEMENT) {
+        case 2:
+          if (view instanceof Int16Array)
+            return i => dv.getInt16(i * 2, true)
+          if (view instanceof Uint16Array)
+            return i => dv.getUint16(i * 2, true)
+          break;
+        case 4:
+          if (view instanceof Int32Array)
+            return i => dv.getInt32(i * 4, true)
+          if (view instanceof Uint32Array)
+            return i => dv.getUint32(i * 4, true)
+          if (view instanceof Float32Array)
+            return i => dv.getFloat32(i * 4, true)
+          break;
+        case 8:
+          if (view instanceof BigInt32Array)
+            return i => dv.getBigInt32(i * 8, true)
+          if (view instanceof BigUint32Array)
+            return i => dv.getBigUint32(i * 8, true)
```

Comment:
I assume these were supposed to be BigInt**64**. The tests should've caught these typos, worth extending to cover these cases too.

Reply:
> worth extending to cover these cases too.

If there is something that doesn't yet/still work on bigendian with this PR, it is fine to not fix it here, but leave it for a later PR. This is a large PR already, so to ease development and allow a comfortable iteration workflow, it's ok to land this in the state it is, and look at improving the suite to cover more cases in a followup.

I ran the tests on bigendian0 suite on this PR to see that it is green, so that's a good first step.

Reply:
> If there is something that doesn't yet/still work on bigendian with this PR, it is fine to not fix it here, but leave it for a later PR.

As far as I can tell, this is something that worked before this PR and gets broken by the change due to typos I commented on, we just don't have tests to catch this regression.

That's different from, say, leaving new functionality for later.